Error in TypeScript types when using NodeNext module mode

I'm in the process of migrating a project that uses `@github/time-elements` to ESM. However, when I try to compile my project, I get the following error from `tsc`:

```
node_modules/@github/time-elements/dist/index.d.ts:1:30 - error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './local-time-element.js'?

1 import LocalTimeElement from './local-time-element';
                               ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@github/time-elements/dist/index.d.ts:2:33 - error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './relative-time-element.js'?

2 import RelativeTimeElement from './relative-time-element';
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@github/time-elements/dist/index.d.ts:3:28 - error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './time-ago-element.js'?

3 import TimeAgoElement from './time-ago-element';
                             ~~~~~~~~~~~~~~~~~~~~

node_modules/@github/time-elements/dist/index.d.ts:4:30 - error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './time-until-element.js'?

4 import TimeUntilElement from './time-until-element';
```

I built a minimal reproduction in the following repo: https://github.com/nwalters512/github-time-elements-typescript-bug

Because this package declares itself as a module (it uses `"type": "module"` in `package.json`), its published type declarations must use file extensions.

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,6 +1,6 @@
-import LocalTimeElement from './local-time-element'
-import RelativeTimeElement from './relative-time-element'
-import TimeAgoElement from './time-ago-element'
-import TimeUntilElement from './time-until-element'
+import LocalTimeElement from './local-time-element.js'
+import RelativeTimeElement from './relative-time-element.js'
+import TimeAgoElement from './time-ago-element.js'
+import TimeUntilElement from './time-until-element.js'
 
 export {LocalTimeElement, RelativeTimeElement, TimeAgoElement, TimeUntilElement}

